diff --git a/cmake/draco_targets.cmake b/cmake/draco_targets.cmake
index c8c79f5..841235c 100644
--- a/cmake/draco_targets.cmake
+++ b/cmake/draco_targets.cmake
@@ -162,7 +162,7 @@ macro(draco_add_executable)
   endif()

   if(exe_LIB_DEPS)
-    if(CMAKE_CXX_COMPILER_ID MATCHES "^Clang|^GNU")
+    if((CMAKE_CXX_COMPILER_ID MATCHES "^Clang|^GNU") AND NOT APPLE)
       # Third party dependencies can introduce dependencies on system and test
       # libraries. Since the target created here is an executable, and CMake
       # does not provide a method of controlling order of link dependencies,
