From 40ec4060bd8afc86b5149078aa3f3faa2b07f8fa Mon Sep 17 00:00:00 2001
From: Charles Milette <charles.milette@gmail.com>
Date: Tue, 14 Apr 2026 23:42:29 -0400
Subject: [PATCH] Remove version constraint on GDAL

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index a653084..94b744b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -52,7 +52,7 @@ endif()
 include(GetGitRevisionDescription)
 get_git_head_revision(GIT_REFSPEC GIT_SHA1)
 
-find_package(GDAL 1.9.0 REQUIRED)
+find_package(GDAL REQUIRED)
 
 
 ###############################################################################
-- 
2.52.0.windows.1

