diff --git a/CMakeLists.txt b/CMakeLists.txt
index 948b604..a6e2da3 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -26,7 +26,7 @@ if(OPENMP_FOUND)
 endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX)
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -march=native -Wall -Wextra -Wundef -pedantic")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wundef -pedantic")
     if( (NOT ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "aarch64") AND (NOT ${CMAKE_SYSTEM_PROCESSOR} STREQUAL "arm64") )
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mfpmath=sse -funroll-loops")
     endif()
diff --git a/elibs/CMakeLists.txt b/elibs/CMakeLists.txt
index 8128284..c4c1d31 100644
--- a/elibs/CMakeLists.txt
+++ b/elibs/CMakeLists.txt
@@ -22,8 +22,8 @@ externalproject_add(ext_rayint
 
 externalproject_add(ext_eigen
     PREFIX          ext_eigen
-    URL             https://gitlab.com/libeigen/eigen/-/archive/3.3.2/eigen-3.3.2.tar.gz
-    URL_MD5         02edfeec591ae09848223d622700a10b
+    URL             https://gitlab.com/libeigen/eigen/-/archive/3.4.0/eigen-3.4.0.tar.gz
+    URL_MD5         4c527a9171d71a72a9d4186e65bea559
     SOURCE_DIR      ${CMAKE_SOURCE_DIR}/elibs/eigen
     CONFIGURE_COMMAND ""
     BUILD_COMMAND   ""
