diff --git a/vcg/complex/algorithms/update/selection.h b/vcg/complex/algorithms/update/selection.h
index 48b90d22..f462bc78 100644
--- a/vcg/complex/algorithms/update/selection.h
+++ b/vcg/complex/algorithms/update/selection.h
@@ -106,7 +106,7 @@ public:
     fsHandle fsH = fsV.back();
     tsHandle tsH = tsV.back();
 
-    if(! (Allocator<ComputeMeshType>::template IsValidHandle(*_m, vsH))) return false;
+    if(! (Allocator<ComputeMeshType>::template IsValidHandle<bool>(*_m, vsH))) return false;
 
     for(auto vi = _m->vert.begin(); vi != _m->vert.end(); ++vi)
       if( !(*vi).IsD() )
